refactor(span): use `Hasher` from `std`

overlookmotel opened the pull request:
Use `std::hash::Hasher` instead of `core::hash::Hasher`.

I don't actually know if importing `core` as well as `std` has any compile time perf impact, but as we're using `std` anyway, we may as well use it for everything.

Import into the `auto_impl_content_hash` module to make the macro a little easier to read.

overlookmotel commented:
* **#5476** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5476?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
This is just nits. Not really important.

DonIsaac commented on crates/oxc_span/src/hash.rs:
```diff
@@ -53,11 +52,12 @@ impl<'a, T: ContentHash> ContentHash for oxc_allocator::Vec<'a, T> {
             $(
                 impl ContentHash for $t {
-                    fn content_hash<H: core::hash::Hasher>(&self, state: &mut H) {
+                    fn content_hash<H: std::hash::Hasher>(&self, state: &mut H) {
```
```suggestion
                    fn content_hash<H: Hasher>(&self, state: &mut H) {
```

overlookmotel committed:
refactor(span): use `Hasher` from `std` (#5476)

Use `std::hash::Hasher` instead of `core::hash::Hasher`.

I don't actually know if importing `core` as well as `std` has any compile time perf impact, but as we're using `std` anyway, we may as well use it for everything.

Import into the `auto_impl_content_hash` module to make the macro a little easier to read.
